File: src/main/java/com/elchologamer/userlogin/manager/LocationsManager.java
package com.elchologamer.userlogin.manager;

import com.elchologamer.userlogin.UserLogin;
import com.elchologamer.userlogin.util.CustomConfig;
import org.bukkit.Location;
import org.bukkit.World;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;

public class LocationsManager {

    private final UserLogin plugin = UserLogin.getPlugin();
    private final CustomConfig locations = new CustomConfig("locations.yml", false);

    public void savePlayerLocation(Player player) {
        saveLocation("playerLocations." + player.getUniqueId(), player.getLocation());
    }

    public void saveLocation(String key, Location location) {
        ConfigurationSection section = locations.get().createSection(key);

        section.set("world", location.getWorld().getName());
        section.set("x", location.getX());
        section.set("y", location.getY());
        section.set("z", location.getZ());
        section.set("yaw", location.getYaw());
        section.set("pitch", location.getPitch());

        locations.save();
    }

    public Location getPlayerLocation(Player player, Location def) {
        Location loc = getPlayerLocation(player);
        return loc != null ? loc : def;
    }

    public Location getPlayerLocation(Player player) {
        return getLocation("playerLocations." + player.getUniqueId());
    }

    public Location getLocation(String key, Location def) {
        Location loc = getLocation(key);
        return loc != null ? loc : def;
    }

    public Location getLocation(String key) {
        ConfigurationSection section = locations.get().getConfigurationSection(key);
        if (section == null) return null;

        // Get world
        World world = plugin.getServer().getWorld(section.getString("world", ""));
        if (world == null) world = plugin.getServer().getWorlds().get(0);

        Location spawn = world.getSpawnLocation();

        return new Location(
                world,
                section.getDouble("x", spawn.getX()),
                section.getDouble("y", spawn.getY()),
                section.getDouble("z", spawn.getZ()),
                (float) section.getDouble("yaw", spawn.getYaw()),
                (float) section.getDouble("pitch", spawn.getPitch())
        );
    }

    public void reload() {
        locations.reload();
    }

}

File: src/main/kotlin/com/elchologamer/userlogin/manager/LocationsManager.kt
package com.elchologamer.userlogin.manager

import com.elchologamer.userlogin.UserLogin.Companion.plugin
import com.elchologamer.userlogin.util.CustomConfig
import org.bukkit.Location
import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player

class LocationsManager {
    val locations = CustomConfig("locations.yml")

    fun savePlayerLocation(player: Player) {
        saveLocation("playerLocations." + player.uniqueId, player.location)
    }

    fun saveLocation(key: String, location: Location) {
        val section = locations.config.createSection(key)

        section["world"] = location.world!!.name
        section["x"] = location.x
        section["y"] = location.y
        section["z"] = location.z
        section["yaw"] = location.yaw
        section["pitch"] = location.pitch

        locations.save()
    }

    fun getPlayerLocation(player: Player, def: Location): Location {
        return getPlayerLocation(player) ?: def
    }

    fun getPlayerLocation(player: Player): Location? {
        return getLocation("playerLocations." + player.uniqueId)
    }

    fun getLocation(key: String, def: Location): Location = getLocation(key) ?: def

    fun getLocation(key: String): Location? {
        val section: ConfigurationSection = locations.config.getConfigurationSection(key) ?: return null

        // Get world
        val worldName = section.getString("world") ?: ""
        val world = plugin.server.getWorld(worldName) ?: plugin.server.worlds[0]

        val spawn = world.spawnLocation

        // Return location
        return Location(
            world,
            section.getDouble("x", spawn.x),
            section.getDouble("y", spawn.y),
            section.getDouble("z", spawn.z),
            section.getDouble("yaw", spawn.yaw.toDouble()).toFloat(),
            section.getDouble("pitch", spawn.pitch.toDouble()).toFloat()
        )
    }

}